scio-core/src/main/scala/com/spotify/scio/ScioResult.scala

```diff
@@ -65,19 +68,59 @@ abstract class ScioResult private[scio] (val internal: PipelineResult) {
   /** Pipeline's current state. */
   def state: State = Try(internal.getState).getOrElse(State.UNKNOWN)
 
-  /** Save metrics of the finished pipeline to a file. */
-  def saveMetrics(filename: String): Unit = {
+  /** Save metrics of the finished pipeline to a file or directory. */
+  def saveMetrics(path: String): Unit = {
+    val isDirectory = path.endsWith(File.separator)
+
+    val resourceId = if (isDirectory) {
+      val randomFileName = f"${UUID.randomUUID()}.json"
```

Comment:
hm, do we actually need this to be random? mould just `metrics.json` work? or even `metrics-{jobId}.json` ?

Reply:
maybe appending timestamp can be good?